src/main/scala/org/spartanz/parserz/ParsersModule.scala

```diff
@@ -283,6 +284,24 @@ trait ParsersModule extends ExprModule {
           }
         }
 
+      case Grammar.GADT.ZipUnsafe(gs) =>
+        (s: S, in: Input) => {
+          val size = gs.length
+          val arr: Array[Any] = Array.ofDim(size)
+          var i = 0
+          var state: S = s
+          var input: Input = in
+          var res: E \/ (Input, Array[Any]) = Right((input, arr))
+          while (i < size && res.isRight) {
+            val (s1, res1): (S, E \/ (Input, Any)) = parser(gs(i))(state, input)
+            res1.foreach { case (i1, v1) => input = i1; arr(i) = v1 }
+            state = s1
+            res = res1.map(_ => (input, arr))
+            i += 1
+          }
+          (state, res)
+        }
```

Comment:
I have added `ZipUnsafe` into existing algebra for demo.
Correct implementation is to have `Grammar.GADT` compiled into another low level algebra, which in turn will be used to produce parser/printer functions.